### PR TITLE
doc, vm, template

### LINKED ISSUE #62: fix submission for M2
Response from the reviewer:
```
Thanks, FP, for a precise explanation, I am with you (satisfied) on 1) module(s) source and 2) A CI job running test(s).

The remaining task is to provide constructive 3) documentation: "Documentation is updated, readable and reflects the processed services." in the resubmission PoA-2. Thus, an average Linux/Cloud-based knowledge person can follow, replicate and implement your team/proposal's work/outputs up to this milestone 2 and non-nix nor computing literacy community members, and mainly voters, are able to simplify and understand "Streamlining Cardano Deployment with Enhanced NixOS Module" proposal progress to this point as a fully open-source funded proposal.

Thank you.
```
Andrea's comments:
So basically the only two remaining things to do are docs:

    docs for non nix savy people (but with experience in linux/cloud). I would say to add a page to the website recapping where to download NixOS, and how to add our modules to a standard configuration (assuming flakes). IMO it should be as minimal as possible with all the needed references to manual and official docs.
    an explanation of what cardano.nix aims to do, I would say we can expand a bit the README and/org the [main website page](https://mlabs-haskell.github.io/cardano.nix/). IMO what they want is a paragraph where we assume that readers already know Cardano but not NixOS, so we can explain why NixOS is cool and how easily you can declare what you want (e.g. a cardano node) instead of doing imperative non-reprodubile steps to get the same result.

### PR DESCRIPTION
**Pull request by brainrake**
- add documentation: intro, installation, running vm, deploy
- add vm in `nixosConfiguration` and `apps`
- add flake template

fix #62
related #41

### CODE REVIEW

File: docs/getting-started/deploy.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+- a [https://zero-to-nix.com/concepts/nixos#configuration](NixOS configuration) to run cardano services, under `nixosConfigurations.server`
+- an app to run the virtual machine as above, under `apps.x86_64-linux.vm`
+
+The file `configuration.nix` is the configuration for the machine.
+
+### Customize
+
+To learn more, browse available [NixOS options in nixpkgs](https://search.nixos.org/options) and [NixOS options provided by cardano.nix](https://mlabs-haskell.github.io/cardano.nix/reference/module-options/cardano/) (see other modules in the menu on the left).
+
+Add these options to `configuration.nix` to
```
Comment (aciceri) [inline docs/getting-started/deploy.md:38]: What did you want to write here?

File: template/vm.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  ...
+}: {
+  imports = [
+    (modulesPath + "/virtualisation/qemu-vm.nix")
+    (modulesPath + "/profiles/qemu-guest.nix")
+  ];
+
+  # WARNING: don't use this in production
+  # Allow root login without password
+  users.users.root.hashedPasswordFile = lib.mkOverride 150 "${pkgs.writeText "hashed-password.root" ""}";
```
Comment (aciceri) [inline template/vm.nix:14]: As above

**Review by aciceri — COMMENTED** [review]
I directly addressed some small things in #66

File: template/vm.nix (hunk 2 above)
Comment (brainrake) [reply template/vm.nix]: fixed

File: configurations/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        {nixpkgs.hostPlatform = "x86_64-linux";}
+      ];
+    };
+  };
+  perSystem = _: {
+    apps = {
+      vm-preview = {
+        type = "app";
+        program = "${self.nixosConfigurations.vm-preview.config.system.build.vm}/bin/run-nixos-vm";
+      };
```
[the remark this answers is not in the document]
Comment (brainrake) [reply configurations/default.nix:21]: added

**Pull request merged**